webapp/lib/app/nook/controller.dart

```diff
@@ -1023,6 +1036,47 @@ class NookController extends Controller {
               .messageViewWithId(messageTagData.messageId)
               .removeTag(messageTagData.tagId);
           }, onError: showAndLogError);
+        if (messageTagData.tagId == selectedMessageTagId) {
+          actionObjectState = null;
+          selectedMessageTagId = null;
+          selectedTagMessageId = null;
+        }
+        break;
+      case UIAction.selectMessageTag:
+        command(UIAction.deselectConversationSummary, null);
+        command(UIAction.deselectConversationTag, null);
+        command(UIAction.deselectMessage, null);
```

Comment:
In testing it seems that one can "select" multiple tags if they're on different messages in the same conversation (only in the view though, the controller keeps track correctly). I think it's because we should be having here a call to `command(UIAction.deselectMessageTag, null);`  as the line `messageViewWithId().markSelectedTag()` below doesn't actually deselect tags on other messages.

So let's extract these 4 `deselect...` methods into a helper method 'deselectAllConversationElements` and call it at the beginning of all `select...` methods. Then we can also simplify `markTagSelected` as below

```suggestion
        command(UIAction.deselectMessage, null);
        command(UIAction.deselectMessageTag, null);
```

Reply:
Ah good catch - I tested by selecting only with in the message and hence didn't catch this. Moved this to a separate function and called it across all the places where we select the elements.